src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -329,16 +331,18 @@ public AgentSettings LoadSettings()
             bool runAsService = command.GetRunAsService();
             if (runAsService)
             {
-                if (!new WindowsPrincipal(WindowsIdentity.GetCurrent()).IsInRole(WindowsBuiltInRole.Administrator))
-                {
-                    Trace.Error("Needs Administrator privileges for configure agent as windows service.");
-                    throw new SecurityException(StringUtil.Loc("NeedAdminForConfigAgentWinService"));
-                }
-
                 Trace.Info("Configuring to run the agent as service");
                 var serviceControlManager = HostContext.GetService<IWindowsServiceControlManager>();
                 serviceControlManager.ConfigureService(agentSettings, command);
             }
+            //This will be enabled with AutoLogon code changes are tested
+            else
+            {                
+                ConfigureAutoLogonIfNeeded(command);
+                //Imp: The machine may restart if the autologon user is not same as the current user
```

Comment:
imp meant important. But now I have removed the redirection. It is 'Important' now. :)

---
In reply to: [118601792](https://github.com/Microsoft/vsts-agent/pull/985#discussion_r118601792) [](ancestors = 118601792)